src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -156,23 +161,29 @@ public DynamicObject readString(DynamicObject pointer, int length) {
     public static abstract class PointerReadStringToNullPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isNullPointer(pointer)")
-        public DynamicObject readNullPointer(DynamicObject pointer) {
-            return createString(RopeConstants.EMPTY_ASCII_8BIT_ROPE);
+        public DynamicObject readNullPointer(DynamicObject pointer,
+                                             @Cached("create()") AllocateObjectNode allocateObjectNode) {
+            return allocateObjectNode.allocate(coreLibrary().getStringClass(), Layouts.STRING.build(false, false, RopeConstants.EMPTY_ASCII_8BIT_ROPE));
```

Comment:
I haven't done any work to make creating strings with an existing rope any easier. This is all work around creating strings from `byte[]` or `java.lang.String`.